src/main/java/com/google/devtools/build/lib/analysis/test/TestActionBuilder.java

```diff
@@ -386,6 +386,14 @@ private TestParams createTestAction(int shards)
       testRunfilesSupplier = SingleRunfilesSupplier.create(runfilesSupport);
     }
 
+    ActionOwner actionOwner =
+        testConfiguration.useTargetPlatformForTests()
+            ? ruleContext.getTestActionOwner()
+            : getOwner();
+    if (actionOwner == null) {
```

Comment:
Since `RuleContext.getTestActionOwner()` can't ever return null, I think this should be cleaned up.

So:
1. Can all the logic be moved into `TestActionBuilder.getOwner`? Or is that called from too many places?
2. Can all the logic be moved into `RuleContext.getTestActionOwner`?

Reply:
Done!